### DISCUSSION TITLE
odbclient - support for shell and beam elements

### DISCUSSION THREAD

**Discussion by MaximLutovinov** [boschresearch/pylife#63] [Ideas]
Hello, 

I've encountered a problem when reading odb files - the current version of odbclient does not support shell and beam elements. Besides solid elements, I always have both - shells and beams - in odbs I'm working with. Sometimes also connectors (conn3d2 elements).

If you ever have time, it would be amazing to update odb client so it works with shells too or ignore elements it is not able to work with.

![image](https://github.com/boschresearch/pylife/assets/154364083/5b189d57-a024-4d7b-9239-f2ddf5b700b3)

**Comment by johannes-mueller** [comment]
That would be a nice enhancement indeed. I would welcome if someone with stake in this would take this on as we cannot prioritize it short-term.

I would like use this thread to collect thoughts and material on how to do that exactly, so please post links to any helpful material.

## Stress tensor

In [this posting](https://github.com/boschresearch/pylife/discussions/58#discussioncomment-8053604) it seems as if the stress tensor consists only of four components instead of the usual six (`S11`, `S22`, `S33`, `S12`, `S13`, `S23`).

### Questions

* How is the stress tensor in Abaqus beam and shell elements exactly defined?
* What assumptions (like e.g. plain stress, plain strain) do we need to map those to the usual 3D stress tensor.
